services/terrain/src/terrain/services/metadata/apps.clj

```diff
@@ -48,13 +48,14 @@
   (assert-valid user-agent "Missing or empty request parameter: user-agent")
   (let [username    (:username current-user)
         user        (:shortUsername current-user)
-        workspace   (dm/get-workspace)
-        preferences (user-prefs (:username current-user))
-        login-time  (:login_time (dm/record-login ip-address user-agent))]
+        workspace   (future (dm/get-workspace))
+        preferences (future (user-prefs (:username current-user)))
```

Comment:
There's actually a lot buried in this `user-prefs` call, since it's also in charge of determining if output directories need to be created/doing so/etc.